src/org/sosy_lab/java_smt/basicimpl/AbstractProver.java

```diff
@@ -134,6 +267,25 @@ protected ImmutableSet<T> getAssertedConstraintIds() {
     return builder.build();
   }
 
+  protected void checkInterpolationArguments(Collection<T> formulasOfA) {
+    checkArgument(
+        getAssertedConstraintIds().containsAll(formulasOfA),
+        "Interpolation can only be done over previously asserted formulas. %s",
+        MoreStrings.lazyString(() -> getErrorString(formulasOfA)).toString());
```

Comment:
Calling `toString()` forces the lazy string to be eagerly evaluated. I would simply use an `if` branch and `throw`. Not everything needs to use fancy features.